common/predictive-text/worker/index.ts

```diff
@@ -30,6 +30,7 @@
  */
 
 /// <reference path="../message.d.ts" />
+/// <reference path="../../../web/source/text/kmwstring.ts" />
```

Comment:
We could move kmwstring to common/engine/web perhaps? That avoids the cross-linking issues we see here? Mind you, it's only a reference, not a link.

Reply:
> > Is this something we'd be interested in doing for KMW as a whole, eventually? If so, I'd rather go ahead and use that now for the LMLayer.
> 
> I'd seriously prefer not to switch to a new low-level API 2 weeks before beta. Let's look at changing this over in 14.0, perhaps. It's not urgent -- the existing code works and is solidly tested, and the likelihood of conflict is very low given the `kmw-` prefix we use on the functions on `String`, so it's really a case of it "ain't broke" even though it lacks purity.

I'd like to point out that before this PR, LMLayer code never touched or utilized kmwstring.ts; this is why the addressed bug currently exists.  There's no "changing out" to be done in any pre-existing code; it's all contained within this PR.  

The one legitimate concern I can see would be if the suggested library has compatibility issues with old JS engines (like Android API 19).  Otherwise, I feel turning kmwstring.ts into a library by moving it in this manner is more trouble than it's worth.

Reply:
There is considerable risk in assuming that all the functions in UTFString work identically to those in kmwstring. kmwstring is a known quantity; UTFString is probably fine but is unknown to us at this time.

We agreed in our review meeting to use kmwstring here, for now, and look at moving to say UTFString across all of KMW and LMLayer in 14.0, once we have had time to validate consistency between the two libraries.